utils/src/treev2/quickPickWizard/QuickPickWithCreateStep.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../../index';
+import { getLastNode } from './QuickPickWizardContext';
+import { ContextValueFilterQuickPickOptions, ContextValueQuickPickStep } from './ContextValueQuickPickStep';
+import { localize } from '../../localize';
+import { NoResourceFoundError } from '../../errors';
+
+type CreateCallback = <TNode extends unknown>() => TNode | Promise<TNode>;
+interface CreateQuickPickOptions extends ContextValueFilterQuickPickOptions {
+    skipIfOne?: never; // Not allowed in CreateQuickPickStep
+    createLabel?: string;
+    createCallback: CreateCallback;
+}
+
+export class CreateQuickPickStep<TContext extends types.QuickPickWizardContext> extends ContextValueQuickPickStep<TContext, CreateQuickPickOptions> {
+    public override async prompt(wizardContext: TContext): Promise<void> {
+        await super.prompt(wizardContext);
+
+        const lastNode = getLastNode<unknown | CreateCallback>(wizardContext);
+        if (typeof lastNode === 'function') {
+            // If the last node is a function, pop it off the list and execute it
+            const callback = wizardContext.pickedNodes.pop() as unknown as CreateCallback;
+            wizardContext.pickedNodes.push(await callback());

```

Comment:
Definitely out of the scope of this PR, but I think ideally, we wouldn't await the callback here, but would instead only prompt, store those values in context, and then execute and set the remaining contexts as needed during execution.

Reply:
Yeah that's exactly the experience we want.